dulwich/repo.py

```diff
@@ -146,7 +146,8 @@ def _get_default_identity() -> Tuple[str, str]:
         except KeyError:
             fullname = None
         else:
-            fullname = gecos.split(",")[0]
+            if not gecos:
```

Comment:
Is this tested? It seems like this is the wrong way around? Also, you'd probably want to assign fullname even if gecos is empty.